frontend/src/components/Admin/CoffeeChats/CoffeeChatDashboard.tsx

```diff
@@ -0,0 +1,138 @@
+import { useEffect, useMemo, useState } from 'react';
+import { Header, Loader, Button, Table } from 'semantic-ui-react';
+import { ExportToCsv, Options } from 'export-to-csv';
+import styles from './CoffeeChatDashboard.module.css';
+import CoffeeChatAPI from '../../../API/CoffeeChatAPI';
+import { useMembers } from '../../Common/FirestoreDataProvider';
+
+type CoffeeChatStats = {
+  fulfilledCategories: string[];
+  bingo: boolean;
+  blackout: boolean;
+};
+
+const CoffeeChatDashboard: React.FC = () => {
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [bingoBoard, setBingoBoard] = useState<string[][]>([]);
+  const [coffeeChats, setCoffeeChats] = useState<CoffeeChat[]>([]);
+
+  const allMembers = useMembers();
+
+  useEffect(() => {
+    CoffeeChatAPI.getCoffeeChatBingoBoard().then((board) => {
+      setBingoBoard(board);
+    });
+
+    CoffeeChatAPI.getAllCoffeeChats().then((chats) => {
+      setCoffeeChats(chats.filter((chat) => chat.status === 'approved'));
+    });
+
+    setIsLoading(false);
+  }, []);
+
+  const categories = bingoBoard.flat();
+
+  const coffeeChatStats = useMemo(() => {
+    const stats: { [key: string]: CoffeeChatStats } = {};
+
+    allMembers.forEach((member) => {
+      const size = bingoBoard.length;
+      const chatsByMember = coffeeChats.filter((chat) => chat.submitter.netid === member.netid);
+      const fulfilledCategories = chatsByMember.map((chat) => chat.category);
+      const blackout = categories.every((category) => fulfilledCategories.includes(category));
+      let bingo = false;
+
+      const linesToCheck = [
+        ...bingoBoard, // Rows
```

Comment:
It would be nice is if we could unify this logic with what the logic used on the member side that already does these same checks!